### ISSUE TITLE
MiniMaxM2 silently applies full-head RoPE: config drops the checkpoints' legacy `rotary_dim` field

### ISSUE THREAD

**Issue by yuhezhang-ai** [huggingface/transformers#48241]
### System Info

Verified on current `main` (`3d6c720188`, `5.16.0.dev0`); also present in v5.12.1. torch 2.10, linux.

### Reproduction

All released `MiniMaxAI/MiniMax-M2*` checkpoints are trained with **partial RoPE** (`rotary_dim: 64` of `head_dim: 128`) and express it in `config.json` through a legacy `rotary_dim` field — their original `trust_remote_code` config class derived `partial_rotary_factor = rotary_dim / head_dim` in `__init__`. The in-tree `MiniMaxM2Config` keeps `rotary_dim` as an opaque extra attribute and `RopeParametersMixin.convert_rope_params_to_dict` only recognizes the standard `partial_rotary_factor` spelling, so `rope_parameters` never receives a partial factor and `MiniMaxM2RotaryEmbedding` silently rotates **all 128 head dims with a full-width frequency ladder** — both the frequency spacing and the rotate-half pairing are wrong in every attention layer, with no warning.

Config-level repro (no weights needed):

```python
from transformers import MiniMaxM2Config
from transformers.models.minimax_m2.modeling_minimax_m2 import MiniMaxM2RotaryEmbedding

# Mirrors MiniMaxAI/MiniMax-M2* config.json (rotary_dim: 64, head_dim: 128).
config = MiniMaxM2Config(rotary_dim=64)
print(config.rope_parameters)   # {'rope_theta': 5000000.0, 'rope_type': 'default'}  <- no partial factor
print(MiniMaxM2RotaryEmbedding(config).inv_freq.shape)
# torch.Size([64]) -> full-head rotation. Expected torch.Size([32]) for rotary_dim=64.
```

Or directly from the released checkpoint's metadata:

```python
from transformers import AutoConfig
config = AutoConfig.from_pretrained("MiniMaxAI/MiniMax-M2")
print(config.rotary_dim, config.head_dim, config.rope_parameters)
# 64 128 {'rope_theta': 5000000, 'rope_type': 'default'}
```

### Measured impact

We hit this in NVIDIA NeMo AutoModel's checkpoint-parity CI while using the in-tree implementation as the vanilla reference for `MiniMaxAI/MiniMax-M2.7`:

- Full-model logits vs a faithful implementation of the checkpoint's own modeling semantics on a 2,048-token document: **mean KL 13.1, p95 22.4, cosine 0.02** — the model is effectively destroyed while loading cleanly.
- Minimal causal proof: on a tiny randomly-initialized 2-layer model in fp32, the in-tree forward diverges from the reference (cosine 0.885); adding `"partial_rotary_factor": 0.5` to `rope_parameters` on the same config makes the *unmodified* in-tree model match the reference to max-abs 7e-7.

### Expected behavior

`MiniMaxM2Config(rotary_dim=64)` should yield `rope_parameters["partial_rotary_factor"] == 0.5` so the model rotates only the trained 64 dims — matching the checkpoints' original remote-code config. Fix (following the `GPTNeoXConfig` `rotary_pct` precedent): map the legacy field in a `convert_rope_params_to_dict` override. PR incoming.
